packages/datadog-plugin-aws-sdk/src/services/lambda.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const log = require('../../../dd-trace/src/log')
+
+class Lambda {
+  generateTags (params, operation, response) {
+    const tags = {}
+
+    if (!params || !params.FunctionName) return tags
+
+    return Object.assign(tags, {
+      'resource.name': `${operation} ${params.FunctionName}`,
+      'aws.lambda': params.FunctionName
+    })
+  }
+
+  requestInject (span, request, tracer) {
+    const operation = request.operation
+    if (operation === 'invoke') {
+      if (!request.params) {
+        request.params = {}
+      }
+
+      const isSyncInvocation = !request.params.InvocationType ||
+        request.params.InvocationType === 'RequestResponse'
+
+      if (isSyncInvocation) {
+        try {
+          const _datadog = {}
+          tracer.inject(span, 'text_map', _datadog)
+          if (!request.params.ClientContext) {
+            const context = { custom: { _datadog } }
+            request.params.ClientContext = Buffer.from(JSON.stringify(context)).toString('base64')
+          } else {
+            const existingContextJson = Buffer.from(request.params.ClientContext, 'base64').toString('utf-8')
```

Comment:
Is there a way to catch the JSON before so we don't have to decode it? Generally speaking we try to avoid having to decode/re-encode user data when we can to avoid adding too much overhead.

Reply:
There is not as this is responsibility of the client. https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Lambda.html#invoke-property